rest/server_context.go

```diff
@@ -628,11 +618,60 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(ctx context.Context, config
 		if resyncRequired {
 			collectionsRequiringResync = append(collectionsRequiringResync, base.ScopeAndCollectionName{Scope: base.DefaultScope, Collection: base.DefaultCollection})
 		}
-		if err := initDataStore(bucket.DefaultDataStore(), db.IndexesAll); err != nil {
+		if err := initDataStore(bucket.DefaultDataStore(), db.IndexesAll, base.DefaultScopeAndCollectionName()); err != nil {
 			return nil, err
 		}
 	}
 
+	startOffline := base.BoolDefault(config.StartOffline, false)
+	var dbInitDoneChan chan error
+	// Initialize any required indexes
+	if len(collectionsRequiringIndexes) > 0 {
+		gsiSupported := bucket.IsSupported(sgbucket.BucketStoreFeatureN1ql)
+		if !gsiSupported {
+			return nil, errors.New("Sync Gateway was unable to connect to a query node on the provided Couchbase Server cluster.  Ensure a query node is accessible, or set 'use_views':true in Sync Gateway's database config.")
```

Comment:
is it worth suggesting only a query node here, or checking that there's only a default collection before suggesting views?

I'd love to _not_ suggest views here.

Or should we move this to a db validation, or do we not talk to server then?

Reply:
This is an existing error message that just got moved around in this PR, so will probably prefer to leave this unchanged for now.